### DISCUSSION TITLE
Recommend Easy Startup Topics

### DISCUSSION THREAD

**Discussion by sophicsage** [markqvist/Reticulum#29] [General]
Hi all.  I'm new to this i.e. Reticulum.  I do have a background in computer forensics.  However I am not a programmer. I can follow instructions and get stuff off the ground but I don't speak tech at a level many programmers may.  But I am very interested in building a local network in my area.

Status:

I've downloaded the apk for Sideband and have installed it on an Android device.  It's very new of course.  Is messaging possible yet?  After installing the app, what is the next step I need to do, in order to have effective two way communications in the app?  I see I have an LXMF address.  How do I utilize that to get a conversation off the ground?  

Feel free to answer the above, or just feel free to lay out some ground work for those who have more computer aptitude than average...but not necessarily at an uber-techie level (as in, clear instructions for processes without assuming someone inherently understands the side details of what is going on.) 

I would hope this project would be much more successful than those projects that rely on advanced Linux command line just to access data and processes.

**Comment by markqvist** [comment]
We really need some more good, intro-level tutorials and guides. I am working on it :) One suggestion for you would be to start out something like this:

- If you have an extra Android device, install Sideband there as well. Go to settings and find your LXMF address.
- From the first phone, tap the "Add conversation" icon on the "Conversations" screen.
- Type in the LXMF address of the other device, and optionally a nickname. Save it.
- You should now be able to send messages between the two phones. This is running completely locally, so let's extend it.
- On a Raspberry Pi (or any other computer you have) on your WiFi/ethernet network, install [NomadNet](https://github.com/markqvist/NomadNet). In the configuration, enable node hosting. Nomadnet will now act as an LXMF propagation node, and phones can message each other, even when offline.
- In the Reticulum configuration of the computer, add an interface to connect to the public RNS Testnet, see [this post](https://github.com/markqvist/Reticulum/discussions/26). You can now message anyone globally that is connected to the testnet, and they can reach you. Be sure to send an announce via the "Announce" icon in the Conversations screen, or configure Sideband to send an announce on app startup.
- You can try sending me a message on the LXMF adrress <0a59377daec8eb8e9c3a>. The most likely path that will take is to go over your local network, then over the Internet to the testnet node in Frankfurt, then to a local Internet-connected node I control, then over a LoRa link to a system that is disconnected from the Internet, and then over a local WiFi network to the device I currently have that LXMF address on.
- You could also add a [LoRa radio](https://unsigned.io/private-messaging-over-lora/) to the Raspberry Pi (or computer), to extend your network locally over LoRa. Repeat this step at a friends house, and you can now message each other in Sideband completely without using the Internet. Reticulum will just route everything over the LoRa interface.
- Or add a connection to the testnet over the Internet at your friends house as well, and you would normally be messaging each other over the Internet. Should an outage occur, Reticulum would start routing your messages over the LoRa part of the network instead.

I know all of the above needs expansion into posts and tutorials that are easier to digest for beginners, but I hope it can at least give you some ideas.

**Comment by markqvist** [comment]
I'll add links to tutorials and other posts for beginners in this thread once they get added.

**Reply by timp34** [reply]
hi @markqvist just a quick question about Lora RNodes and the device signatures.  I have noticed if i switch a LilyGO LoRa32 v2.1 from say a desktop debian computer to a Pi debian install, then the rnodeconf reports the Device signature is not valid. Seems then my lora comms stop.  Is this by design? i.e. to stop impersonation of a rnode etc?  I switched radios by mistake and have chased my tail for a while until a reflashed both radios and they now report "validated local signature"...  and now it all works again...

Cheers

**Reply by markqvist** [reply]
Currently it is only a warning, to let you know the system could not verify the signature and integrity of the device. This is done to encourage flashing your own RNodes so you know they do not contain malicious code.

You can safely ignore this warning if you flashed the RNode yourself, and it still verifies OK on the computer you flashed it from, or if you completely trust the source it comes from.

Reticulum and other tools will also allow you to use it with an unverifiable signature. In the future there will be a configuration option to allow you to use only devices that are verifiable, and an easier way to distribute the public signing keys to your systems, but it is currently not implemented yet.

**Reply by markqvist** [reply]
Btw, if you experience a scenario where comms stop, that is not intended behaviour. If this is the case, I'd very much appreciate if you can outline steps to reproduce the behaviour!

**Reply by timp34** [reply]
thanks for response that makes sense.  I will see if i can reproduce the issue i had before I change anything else!!
